instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RoutesPreviewTest.kt

```diff
@@ -0,0 +1,190 @@
+        mapboxNavigation.setRoutesPreview(initialRoutes)
+        mapboxNavigation.routesPreviewUpdates()
+            .first { it.reason == RoutesPreviewUpdateReasons.PREVIEW_NEW }
+        // start active guidance
+        mapboxNavigation.setNavigationRoutes(currentRoutesPreview!!.routesPreview!!.routesList)
+        mapboxNavigation.setRoutesPreview(emptyList())
+        mapboxNavigation.routeProgressUpdates()
+            .first { it.currentState == RouteProgressState.TRACKING }
+        // preview a different route not leaving action guidance
+        val updatedRoutes = RoutesProvider.dc_very_short_two_legs(activity).toNavigationRoutes()
+        mapboxNavigation.setRoutesPreview(updatedRoutes)
+        mapboxNavigation.routesPreviewUpdates()
+            .first { it.routesPreview?.routesList == updatedRoutes }
+        assertEquals(
+            "active guidance should track initial routes",
+            initialRoutes,
+            currentRoutes?.navigationRoutes
+        )
+        // user decided to switch to previewed routes
+        mapboxNavigation.setNavigationRoutes(currentRoutesPreview!!.routesPreview!!.routesList)
+        mapboxNavigation.setRoutesPreview(emptyList())
+        mapboxNavigation.routeProgressUpdates().first {
+            it.navigationRoute == updatedRoutes[0]
+        }
+    }
+
+    @Test
+    fun start_active_guidance_from_previewed_alternative_route() = sdkTest {
+        // set routes preview
+        val routes = RoutesProvider.dc_short_with_alternative(activity).toNavigationRoutes()
+        mapboxNavigation.setRoutesPreview(routes)
+        val preview = mapboxNavigation.routesPreviewUpdates()
+            .first { it.reason == RoutesPreviewUpdateReasons.PREVIEW_NEW }
+        // switch to alternative route
+        mapboxNavigation.changeRoutesPreviewPrimaryRoute(
+            preview.routesPreview!!.originalRoutesList[1]
+        )
+        val updatedPreview = mapboxNavigation.routesPreviewUpdates()
+            .first { it != preview }
+        // start active guidance
+        mapboxNavigation.setNavigationRoutes(updatedPreview.routesPreview!!.routesList)
+        mapboxNavigation.setRoutesPreview(emptyList())
+        val routesUpdate = mapboxNavigation.routesUpdates()
+            .first { it.reason == RoutesExtra.ROUTES_UPDATE_REASON_NEW }
+        assertEquals(
+            listOf(
+                routes[1],
+                routes[0]
+            ),
+            routesUpdate.navigationRoutes
+        )
+        val previewAlternativeMetadata = updatedPreview.routesPreview!!.alternativesMetadata.first()
+        val activeGuidanceAlternativeMetadata = mapboxNavigation
+            .getAlternativeMetadataFor(routes[0])!!
+//        Doesn't pass as they have different alternative ids
+//        assertEquals(
+//            previewAlternativeMetadata,
+//            activeGuidanceAlternativeMetadata
+//        )
```

Comment:
@DenisPryt , I noticed that metadata which is created using RouteParser.createRoutesData has different alternative id than the one which is created via setRoutes. Do I understand correctly that RouteParser.createRoutesData always creates new id as it probably doesn't track routes from the current navigation session?

Reply:
As far I know `RouteParser.createRoutesData` do not assign identifier to the alternatives. It should be 0.

Reply:
thanks. Added this to the docs and instrumentation tests.